### PR TITLE
Implemented Bean Factory basic logic

### PR DESCRIPTION
**Pull request by DimaGalchenko**

### CODE REVIEW

File: src/main/java/com/codeus/winter/config/DefaultBeanFactory.java [hunk 1]
```diff
@@ -156,33 +161,129 @@ private <T> void checkBeanClassUniqueness(@Nonnull final Class<T> beanClass) {
...
+        if (Boolean.TRUE.equals(initializationStatuses.get(beanName))) {
+            return false;
+        }
+
+        initializationStatuses.put(beanName, false);
+
+        String[] dependsOn = beanDefinition.getDependsOn();
+        if (dependsOn != null) {
+            for (String dependency: dependsOn) {
+                BeanDefinition dependencyBeanDefinition = beanDefinitions.get(dependency);
```
Comment (khshanovskyi) [inline src/main/java/com/codeus/winter/config/DefaultBeanFactory.java]: use Optional instead

Comment (khshanovskyi) [reply src/main/java/com/codeus/winter/config/DefaultBeanFactory.java]: Optional.ofNullable(beanDefinitions.get(dependency)).orElseThrow(() -> BFE(...));

File: src/main/java/com/codeus/winter/config/DefaultBeanFactory.java [hunk 2]
```diff
@@ -156,33 +161,129 @@ private <T> void checkBeanClassUniqueness(@Nonnull final Class<T> beanClass) {
...
+
+        initializationStatuses.put(beanName, true);
+        return true;
     }
 
-    @Override
-    public final boolean containsBeanDefinition(String beanName) {
-        return false;
+    private Object createBeanInstance(String beanName, BeanDefinition beanDefinition) {
+        String className = beanDefinition.getBeanClassName();
```
Comment (khshanovskyi) [inline src/main/java/com/codeus/winter/config/DefaultBeanFactory.java]: use Optional instead

**Review by khshanovskyi — APPROVED** [review]
❤️

**Review by AndriyPavlyuk — APPROVED** [review]

File: src/main/java/com/codeus/winter/config/DefaultBeanFactory.java [hunk 3]
```diff
@@ -31,10 +39,7 @@ public class DefaultBeanFactory implements BeanFactory, BeanDefinitionRegistry {
     public final Object getBean(@Nonnull final String name) throws BeanNotFoundException {
         if (singletonBeans.containsKey(name)) {
             final Object bean = singletonBeans.get(name);
-            postProcessors
-                    .forEach(postProcessor -> postProcessor.postProcessBeforeInitialization(bean, name));
-            postProcessors
-                    .forEach(postProcessor -> postProcessor.postProcessAfterInitialization(bean, name));
+
```
Comment (AndriyPavlyuk) [inline src/main/java/com/codeus/winter/config/DefaultBeanFactory.java]: @DimaGalchenko where do you plan to invoke bean post processors?

File: src/main/java/com/codeus/winter/config/DefaultBeanFactory.java (hunk 1 above)
Comment (DimaGalchenko) [reply src/main/java/com/codeus/winter/config/DefaultBeanFactory.java]: Done

File: src/main/java/com/codeus/winter/config/DefaultBeanFactory.java (hunk 2 above)
Comment (DimaGalchenko) [reply src/main/java/com/codeus/winter/config/DefaultBeanFactory.java]: Done

**Review by Mitos35 — APPROVED** [review]

File: src/main/java/com/codeus/winter/config/DefaultBeanFactory.java (hunk 3 above)
Comment (DimaGalchenko) [reply src/main/java/com/codeus/winter/config/DefaultBeanFactory.java]: Yes, I forgot to put it back. I think it makes sense to invoke bean post processors after factory creates an instance of Bean. And each post processor should be invoked only one time

Comment (DimaGalchenko) [reply src/main/java/com/codeus/winter/config/DefaultBeanFactory.java]: I checked how this works in Spring Context. It's not just calling the bean processor during initialization. I will create a task for this to investigate this issue and implement the solution. Let's leave it as not implemented in scope of this PR.

**Pull request merged**